scipy/stats/tests/test_stats.py

```diff
@@ -2451,12 +2451,12 @@ def test_ttest_ind():
     assert_raises(ValueError, stats.ttest_ind, x, y, nan_policy='raise')
     assert_raises(ValueError, stats.ttest_ind, x, y, nan_policy='foobar')
 
-    try:
-        # test zero division problem
-        t,p = stats.ttest_ind([0,0,0],[1,1,1])
-        assert_equal((np.abs(t),p), (np.inf, 0))
-        assert_equal(stats.ttest_ind([0,0,0], [0,0,0]), (np.nan, np.nan))
+    # test zero division problem
+    t,p = stats.ttest_ind([0,0,0], [1,1,1])
+    assert_equal((np.abs(t),p), (np.inf, 0))
+    assert_equal(stats.ttest_ind([0,0,0], [0,0,0]), (0.0, 1.0))
```

Comment:
I've found part of the discussion by searching through the scipy-dev mailing list : http://article.gmane.org/gmane.comp.python.scientific.devel/16645/

Sorry, I have not looked there before. Haven't been around for so long ;)